### PR TITLE
[Core] Fix incorrect gpu ids if placement group bundle index specified

### LINKED ISSUE #29811: [Core] GPU placement group doesn't honor the bundle index
### What happened + What you expected to happen

For the reproduction script below, I would expect that each bundle is tied to a GPU so the output should be "0, 1, 2, 3" but now it's "0, 0, 1, 1"

### Versions / Dependencies

master

### Reproduction script

```
import ray
from ray.util.placement_group import placement_group
from ray.util.scheduling_strategies import PlacementGroupSchedulingStrategy

ray.init(num_gpus=4)

@ray.remote(num_gpus=0.5, num_cpus=0)
def gpu_task():
    import os
    print(f"task is running on GPU {os.environ['CUDA_VISIBLE_DEVICES']}")

pg = placement_group([{'GPU': 1}] * 4, strategy="SPREAD")
ray.get(pg.ready())
futures = [gpu_task.options(
    scheduling_strategy=PlacementGroupSchedulingStrategy(
        placement_group=pg, placement_group_bundle_index=i)).remote() for i in range(4)]
ray.get(futures)
```

### Issue Severity

_No response_

### PR DESCRIPTION
**Pull request by wuxibin89**
## Why are these changes needed?
This PR fix incorrect `CUDA_VISIBLE_DEVICES` when `placement_group_bundle_index` is specified.

## Related issue number
Closes https://github.com/ray-project/ray/issues/29811

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

cc @jjyao

### CODE REVIEW

**Review by jjyao — COMMENTED** [review]
Nice!

File: python/ray/_private/worker.py [hunk 1]
```diff
@@ -998,7 +998,7 @@ def get_gpu_ids() -> Union[List[int], List[str]]:
     gpu_ids = worker.get_accelerator_ids_for_accelerator_resource(
         ray_constants.GPU, f"^{ray_constants.GPU}_group_[0-9]+_[0-9A-Za-z]+$"
     )
-    if len(gpu_ids) != 0:
```
Comment (jjyao) [inline python/ray/_private/worker.py]: sorry this should be `if gpu_ids:`

Comment (wuxibin89) [reply python/ray/_private/worker.py]: Ah, OK!

**Comment by jjyao** [comment]
Oh actually I think this PR will cause issue if user uses both bundle index and non-bundle index in the same PG.

For example:

```
ray.init(num_gpus=2) # Two GPUs: 0, 1
pg = placement_group([{"GPU": 1}, {"GPU": 1}])

@ray.remote(num_gpus=1)
class Actor:
  pass

actor1 = Actor.options(scheduling_strategy=PlacementGroupSchedulingStrategy(placement_group=pg)).remote()
actor2 = Actor.options(scheduling_strategy=PlacementGroupSchedulingStrategy(placement_group=pg, bundle_index=0)).remote()
```

It's possible that both actors are assigned to GPU 0

**Comment by wuxibin89** [comment]
> Oh actually I think this PR will cause issue if user uses both bundle index and non-bundle index in the same PG.

@jjyao Yes, I can reproduce this problem, I will take a look how to fix it.

**Comment by jjyao** [comment]
@wuxibin89 I think this is not easy to fix right now. This problem will be fixed automatically once we re-implemented PG using labels and virtual clusters (https://github.com/ray-project/enhancements/pull/49)

**Comment by wuxibin89** [comment]
@jjyao Hi, is there any plan that re-implementing PG with labels and virtual clusters? It seems like there's a lot work to do and may not available in short term. We have a project(will open soon) that heavily relay on this fix to colocate multiple actor groups on the same PG.

So can we fix this first and give user a warning not to use bundle index and non-bundle index in the same PG？

**Comment by anyscalesam** [comment]
@wuxibin89  grabbed some time to chat about this further as full fledged VC implementation will be difficult and perhaps overkill.

I think what would make more sense is for us to sit down for 15m and understand your use case well and than brainstorm on a solution that perhaps you can contribute back to.

I setup time for later this week. cc @jjyao

**Pull request closed without merging**